File: guanabara-java/src/com/danillo/algoritmos/lista07/Exercicio68.java
package com.danillo.algoritmos.lista07;

import java.util.Scanner;

public class Exercicio68 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        int mulheresCadastradas = 0, homensMais100Kg = 0;
        float soma = 0, maiorPesoHomens = 0;

        for(int i = 0; i < 8; i++){
            System.out.println("Insira o gênero: ");
            char genero = entrada.next().charAt(0);
            System.out.println("Insira o peso: ");
            float peso = entrada.nextFloat();

            if(genero == 'F'){
                soma = soma + peso;
                mulheresCadastradas++;
            }

            if(genero == 'M' && peso > 100){
                homensMais100Kg++;
            }

            if(genero == 'M' && peso > maiorPesoHomens){
                maiorPesoHomens = peso;
            }
        }

        float mediaPesoMulheres = soma/mulheresCadastradas;

        System.out.println("Mulheres foram cadastradas: " +mulheresCadastradas+
                           "\nHomens com mais de 100kg: " +homensMais100Kg+
                           "\nMédia do peso entre as mulheres: " +mediaPesoMulheres+
                           "\nMaior peso entre os homens: " +maiorPesoHomens);
    }
}

File: guanabara-python/src/lista07/exercicio68.py
mulheres_cadastradas = 0
homens_mais_100kg = 0
soma_peso_mulheres = 0
maior_peso_homens = 0

for i in range(0, 9):
    genero = int(input("Insira o gênero: 1 - Masculino | 2 - Feminino "))
    peso = float(input("Insira o peso: "))

    if genero == 1:
        if peso > maior_peso_homens:
            maior_peso_homens = peso

        if peso > 100:
            homens_mais_100kg = homens_mais_100kg + 1

    elif genero == 2:
        mulheres_cadastradas = mulheres_cadastradas + 1
        soma_peso_mulheres = soma_peso_mulheres + peso

    else:
        print("Insira um gênero válido.")

media_peso_mulheres = soma_peso_mulheres / mulheres_cadastradas

print(f"Mulheres cadastradas: {mulheres_cadastradas}.\nHomens que pesam mais de 100kg: {homens_mais_100kg}."
      f"\nMédia de peso entre as mulheres: {media_peso_mulheres}.\nMaior peso entre os homens: {maior_peso_homens}.")
